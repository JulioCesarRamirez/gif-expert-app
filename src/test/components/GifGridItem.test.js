import React from "react";
import "@testing-library/jest-dom";
import GifGridItem from "../../components/GifGridItem";
import { shallow } from "enzyme";

describe("Test in <GifGridItem />", () => {
  const title = "A title";
  const url = "https://localhost/algo.jpg";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  it("Should show <GifGridItem /> correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("Should has a paragraph with the title", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });
  it("Should has an image similar to url and alt from props", () => {
    const img = wrapper.find("img");
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });
  it("Should has animate__fadeIn className", () => {
    const div = wrapper.find("div");
    const className = div.prop("className");
    expect(className.includes("animate__fadeIn")).toBe(true);
  });
});
