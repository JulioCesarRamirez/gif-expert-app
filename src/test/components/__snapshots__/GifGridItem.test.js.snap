// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <GifGridItem /> Should show <GifGridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="A title"
    src="https://localhost/algo.jpg"
  />
  <p>
    A title
  </p>
</div>
`;
